Current git master fails to build with error: cannot convert ‘QString’ to ‘const QFile&’

```
Gui/Pages.cpp: In member function ‘bool IconsPage::eventFilterIconList(QEvent*)’:
Gui/Pages.cpp:2756:57: error: cannot convert ‘QString’ to ‘const QFile&’
 2756 |                 if (IconManager::instance().addIconFile(path)) {
      |                                                         ^~~~
      |                                                         |
      |                                                         QString
```

I'm building on gentoo with qt 6.9.1

I think it's related to https://github.com/GoldenCheetah/GoldenCheetah/pull/4695
@thejockl : could you have a look?

--- a/src/Gui/Pages.cpp
+++ b/src/Gui/Pages.cpp
@@ -2752,8 +2752,7 @@ IconsPage::eventFilterIconList
             const QList<QUrl> urls = dropEvent->mimeData()->urls();
             int added = 0;
             for (const QUrl &url : urls) {
-                QString path = url.toLocalFile();
-                if (IconManager::instance().addIconFile(path)) {
+                if (IconManager::instance().addIconFile(QFile(url.toLocalFile()))) {
                     ++added;
                 }
             }

